### ISSUE TITLE
Assembly loader event tracing is very expensive (10x application slowdown)

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#78539]
## Repro

1. `dotnet new console` in .NET 7 SDK, copy&paste the following program (repro extracted from actual app where the issue was observed) 
```
using System.Diagnostics;

var sw = new Stopwatch();
for (;;)
{
    sw.Restart();
    for (int i = 0; i < 100000; i++) Type.GetType("System.String, System.Runtime");
    Console.WriteLine(sw.ElapsedMilliseconds);
}
```
2. `dotnet run`
3. Download latest [perfview](https://github.com/microsoft/perfview) and start a global collection session with default settings (Select "Collect / Collect" from menu and press "Start Collection" button)

## Actual result

Program runs more than 10x slower when the perfview collection session is active

## Expected result

Default perfview collection settings should not slow down the program by more than 10%

The problem is caused by very slow assembly loader tracing. The fix should be to either make assembly loader tracing a lot faster and lighter weight for cases like this one and/or assembly loader tracing should not be enabled for default (perfview) event collection settings.

**Comment by vitek-karas** [comment]
I don't think detailed assembly load tracing should be enabled by default - it was not designed for that case. We specifically "Didn't care much" about perf since the goal was to provide on demand diagnostic tool.

@elinor-fung could you please look into disabling this as part of the default collection?

**Comment by elinor-fung** [comment]
Agreed - it is intended for targeted assembly loading investigation and should not be enabled by default.

I'll look at updating perfview / TraceEvent defaults.

**Comment by elinor-fung** [comment]
It looks like some of the VS tools also enable it - I'll get in contact for those as well.
